site/src/hooks/coursebag.ts

```diff
@@ -44,5 +45,13 @@ export function useCoursebag() {
     [addCourseToBag, coursebag, removeCourseFromBag],
   );
 
-  return { coursebag: coursebag ?? [], addCourseToBag, removeCourseFromBag, toggleBookmark };
+  const loadCoursebag = useCallback(async () => {
+    const courseIds = isLoggedIn
+      ? await trpc.savedCourses.get.query()
+      : JSON.parse(localStorage.getItem('coursebag') ?? '[]');
+    const coursebag = await searchAPIResults('courses', courseIds);
+    dispatch(setCoursebag(Object.values(coursebag)));
+  }, [dispatch, isLoggedIn]);
+
+  return { coursebag: coursebag ?? [], addCourseToBag, removeCourseFromBag, toggleBookmark, loadCoursebag };
```

Comment:
instead of returning it as a function, you should just use the `useCourseBag` hook itself to load the user data, similar to how `useLoadTransferredCredits()` works.

Reply:
although, maybe it's still better that the loader is called once. I do like having in a separate file from app.tsx, but i think maybe it's better to have as a different function than useCoursebag...